Added feature with multiflile configuration in ocelot and swagger

dadjh85 opened the pull request:
@Burgyn Here I leave you a pull request of what we have been talking about days ago, sorry for the delay in sending it, but I have been busy for several days. There is a lack of documentation, but if you like the code I will write the documentation because if not, it is not worth to write it.

Burgyn commented:
Hi,
thanks for your PR.
I'll look into it tomorrow.

In the meantime, Can you please try to resolve the merge conflicts?

dadjh85 commented:
> 
> 
> Hi,
> thanks for your PR.
> I'll look into it tomorrow.
> 
> In the meantime, can you please try to resolve the merge conflicts?

I can't click the button to resolve the conflicts, the button appears disabled, you may not have permission for this action.

Burgyn commented on demo/ApiGateway/Startup.cs:
```diff
@@ -59,6 +59,8 @@ public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
                         new KeyValuePair<string, string>("Key2", "Value2"),
                     };
+
+                    //opt.ServerOcelot = "/siteName/apigateway" ;
```
Please remove commented code.

Burgyn commented on src/MMLib.SwaggerForOcelot/Configuration/SwaggerFileConfiguration.cs:
```diff
@@ -0,0 +1,22 @@
+    /// </summary>
+    public class SwaggerFileConfiguration
+    {
+        public List<SwaggerFileRoute> Routes { get; set; } = new List<SwaggerFileRoute>();
```
Please comment all public methods.

Burgyn commented on src/MMLib.SwaggerForOcelot/DependencyInjection/ConfigurationBuilderExtensions.cs:
```diff
@@ -0,0 +1,121 @@
+
+namespace MMLib.SwaggerForOcelot.DependencyInjection
+{
+    public static class ConfigurationBuilderExtensions
```
Please comment public members.

Burgyn commented on src/MMLib.SwaggerForOcelot/DependencyInjection/ConfigurationBuilderExtensions.cs:
```diff
@@ -0,0 +1,121 @@
+            string folder = "/",
+            string fileOfSwaggerEndPoints = SwaggerForOcelotFileOptions.SwaggerEndPointsConfigFile)
+        {
+            List <FileInfo> files = GetListOfOcelotFiles(folder, environment?.EnvironmentName);
```
```suggestion
            List<FileInfo> files = GetListOfOcelotFiles(folder, environment?.EnvironmentName);
```

Burgyn commented on src/MMLib.SwaggerForOcelot/DependencyInjection/ConfigurationBuilderExtensions.cs:
```diff
@@ -0,0 +1,121 @@
+            IEnumerable<FileInfo> ocelotFiles = new DirectoryInfo(folder).EnumerateFiles()
+                                                                         .Where(fi => reg.IsMatch(fi.Name));
+            if (!nameEnvirotment.IsNullOrWhiteSpace())
+                ocelotFiles = ocelotFiles.Where(fi => fi.Name.Contains(nameEnvirotment));
```
Wrap if block with `{}` parentheses.

Burgyn commented on src/MMLib.SwaggerForOcelot/DependencyInjection/ConfigurationBuilderExtensions.cs:
```diff
@@ -0,0 +1,121 @@
+        /// <summary>
+        /// Extension for compatibility of functionality multifile of ocelot
+        /// </summary>
+        /// <param name="builder"></param>
```
Please write comments for all prameters.

Burgyn commented on src/MMLib.SwaggerForOcelot/DependencyInjection/ConfigurationBuilderExtensions.cs:
```diff
@@ -0,0 +1,121 @@
+        /// <param name="folder"></param>
+        /// <param name="fileOfSwaggerEndPoints"></param>
+        /// <returns>a Object IConfigurationBuilder</returns>
+        public static IConfigurationBuilder AddOcelotWithSwaggerSupport(
```
Please write unit test for this method.

Burgyn commented:
> > Hi,
> > thanks for your PR.
> > I'll look into it tomorrow.
> > In the meantime, can you please try to resolve the merge conflicts?
> 
> I can't click the button to resolve the conflicts, the button appears disabled, you may not have permission for this action. i read this text: "Only those with write access to this repository can merge pull requests."

Hi,
yes, button is disabled. But you can merge master branch to your branch and resolve conflicts. Thanks.

Code looks good, I wrote some issues directly to code. Please check it.
Thanks a lot.

dadjh85 commented:
> 
> 
> > > Hi,
> > > thanks for your PR.
> > > I'll look into it tomorrow.
> > > In the meantime, can you please try to resolve the merge conflicts?
> > 
> > 
> > I can't click the button to resolve the conflicts, the button appears disabled, you may not have permission for this action. i read this text: "Only those with write access to this repository can merge pull requests."
> 
> Hi,
> yes, button is disabled. But you can merge master branch to your branch and resolve conflicts. Thanks.
> 
> Code looks good, I wrote some issues directly to code. Please check it.
> Thanks a lot.

Ok, this week, I'll review the changes and update this pull request.

dadjh85 commented:
Hi @Burgyn, I'm updating this pull request with the changes you mentioned and also I've updated the demo project to work with multiple swagger files inside the configuration folder, I've added the different ocelot.SERVICE.json files, I don't have much experience with this kind of tests focused on components, but I've been checking over and maybe I can try to implement these tests, anyway if I have any complication or doubt I'll tell you if you help me to complete or finish them, otherwise you can check the update of this pull request

Burgyn commented:
Hi,

thanks. I will write unit tests later.

One unit test began to fall. Will you look at it?
[ReturnUriFromServiceDiscoveryWhenRouteDoesntExist ](https://github.com/Burgyn/MMLib.SwaggerForOcelot/pull/109/checks?check_run_id=773300238)

dadjh85 commented:
> Hi,
> 
> thanks. I will write unit tests later.
> 
> One unit test began to fall. Will you look at it?

OK, fine, I'll try and look at it today, the test that fails, by tomorrow at the latest I should have it sorted out.

dadjh85 commented:
> Hi,
> 
> thanks. I will write unit tests later.
> 
> One unit test began to fall. Will you look at it?
> [ReturnUriFromServiceDiscoveryWhenRouteDoesntExist ](https://github.com/Burgyn/MMLib.SwaggerForOcelot/pull/109/checks?check_run_id=773300238)

Hi, I've already fixed the test that failed and I've also written the documentation about the developed feature

Burgyn commented:
Hi,

thanks a lot. This feature will be available in version `2.4.0` soon.
